plugins/kubernetes-backend/src/kubernetes-auth-translator/AwsIamKubernetesAuthTranslator.ts

```diff
@@ -28,97 +32,72 @@ export type SigningCreds = {
   sessionToken: string | undefined;
 };
 
+const defaultRegion = 'us-east-1';
+
 /**
  *
  * @public
  */
 export class AwsIamKubernetesAuthTranslator
   implements KubernetesAuthTranslator
 {
-  validCredentials(creds: SigningCreds): boolean {
-    return (creds?.accessKeyId && creds?.secretAccessKey) as unknown as boolean;
-  }
-
-  awsGetCredentials = async (): Promise<Credentials> => {
-    return new Promise((resolve, reject) => {
-      AWS.config.getCredentials(err => {
-        if (err) {
-          return reject(err);
-        }
-
-        return resolve(AWS.config.credentials as Credentials);
-      });
-    });
-  };
-
-  async getCredentials(
-    assumeRole?: string,
-    externalId?: string,
-  ): Promise<SigningCreds> {
-    const awsCreds = await this.awsGetCredentials();
-
-    if (!(awsCreds instanceof Credentials))
-      throw new Error('No AWS credentials found.');
-
-    let creds: SigningCreds = {
-      accessKeyId: awsCreds.accessKeyId,
-      secretAccessKey: awsCreds.secretAccessKey,
-      sessionToken: awsCreds.sessionToken,
-    };
-
-    if (!this.validCredentials(creds))
-      throw new Error('Invalid AWS credentials found.');
-    if (!assumeRole) return creds;
-
-    try {
-      const params: AWS.STS.Types.AssumeRoleRequest = {
-        RoleArn: assumeRole,
-        RoleSessionName: 'backstage-login',
-      };
-      if (externalId) params.ExternalId = externalId;
-
-      const assumedRole = await new AWS.STS().assumeRole(params).promise();
-
-      if (!assumedRole.Credentials) {
-        throw new Error(`No credentials returned for role ${assumeRole}`);
-      }
-
-      creds = {
-        accessKeyId: assumedRole.Credentials.AccessKeyId,
-        secretAccessKey: assumedRole.Credentials.SecretAccessKey,
-        sessionToken: assumedRole.Credentials.SessionToken,
-      };
-    } catch (e) {
-      console.warn(`There was an error assuming the role: ${e}`);
-      throw new Error(`Unable to assume role: ${e}`);
-    }
-    return creds;
-  }
-  async getBearerToken(
+  private async getBearerToken(
     clusterName: string,
     assumeRole?: string,
     externalId?: string,
   ): Promise<string> {
-    const credentials = await this.getCredentials(assumeRole, externalId);
+    const region = process.env.AWS_REGION ?? defaultRegion;
+    let credentials = fromEnv();
+    if (assumeRole) {
+      credentials = fromTemporaryCredentials({
```

Comment:
Does it make sense to integrate with the integration-aws-node package and its `AwsCredentialsManager`? I think the intention was to have this package to have one place with the logic for getting credentials for AWS services. And it was the first place based on v3.

Reply:
@pjungermann I am looking into this know. There is a bit of heavy lifting in this. Requiring a couple of changes to pass down through the provider.